DigitalLearningSolutions.Data/Models/DelegateUpload/DelegateTableRow.cs

```diff
@@ -129,6 +143,18 @@ public bool Validate(IEnumerable<int> allowedJobGroupIds)
             {
                 Error = BulkUploadResult.ErrorReason.TooLongAnswer6;
             }
+            else if (HasPrn.HasValue && HasPrn.Value && string.IsNullOrEmpty(Prn))
+            {
+                Error = BulkUploadResult.ErrorReason.HasPrnButMissingPrnValue;
+            }
+            else if (!string.IsNullOrEmpty(Prn) && (Prn.Length < 5 || Prn.Length > 20))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidPrnLength;
+            }
+            else if (!string.IsNullOrEmpty(Prn) && !PrnRegex.IsMatch(Prn))
+            {
+                Error = BulkUploadResult.ErrorReason.InvalidPrnCharacters;
+            }

```

Comment:
So we don't care if HasPRN is False and a PRN is provided?

Reply:
Unless HasPRN is true we're saving null to the database, so I don't think it needs a validation case.